content/all/react-testing-4-cypress.md

```diff
@@ -15,40 +15,42 @@ tags:
   - msw
 ---
 
-Cypress is a framework-agnostic end-to-end (E2E) testing tool for web apps. Together with Cypress Testing Library, it gives the best test writing experience, and makes applying best practices, we’ve learned in the first article, easy.
+Cypress is a framework-agnostic end-to-end testing (also known as E2E, or integration testing) tool for web apps. Together with Cypress Testing Library and Mock Service Worker, it gives the best test writing experience and makes writing good, resilient to changes, tests straightforward.
 
-**This is the fourth article in the series**, where we learn how to test React apps with Cypress and Cypress Testing Library.
+**This is the fourth article in the series**, where we learn how to test React apps end-to-end using Cypress and Cypress Testing Library, and how to mock network requests using Mock Service Worker.
```

Comment:
Sounds a bit unnatural to me.
Would it make sense to move **end-to-end** before **React apps**?

Reply:
I don't feel that either of the two is more natural ;-/